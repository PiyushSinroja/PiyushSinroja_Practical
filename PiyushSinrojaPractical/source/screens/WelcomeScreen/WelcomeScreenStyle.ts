import {StyleSheet, Dimensions} from 'react-native';
import {globalColors} from '../../constant/Colors';
const {height, width} = Dimensions.get('screen');

/**
 * Welcome Screen Style
 */
export const WelcomeScreenStyle = StyleSheet.create({
  safeArea: {
    backgroundColor: 'white',
    flex: 1,
  },
  userViewContainer: {
    flexDirection: 'column',
    width: '100%',
    marginTop: 40,
  },
  container: {
    backgroundColor: 'white',
    flex: 1,
    alignItems: 'stretch',
    justifyContent: 'space-between',
  },
  inputStyle: {
    alignSelf: 'stretch',
    marginStart: 15,
    marginEnd: 15,
    borderColor: 'gray',
    borderWidth: 1,
    borderRadius: 5,
    marginTop: 0,
    paddingStart: 10,
    fontSize: 14,
    height: height * 0.055,
    justifyContent: 'center',
  },
  saveButtonStyle: {
    color: globalColors.white,
    fontWeight: 'bold'
  }
});
