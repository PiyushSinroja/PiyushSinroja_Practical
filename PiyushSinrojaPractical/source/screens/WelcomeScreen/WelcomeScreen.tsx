const {height, width} = Dimensions.get('screen');
import {
  SafeAreaView,
  View,
  TextInput,
  StyleSheet,
  Dimensions,
  Alert,
  NativeModules,
} from 'react-native';
import {UserActions} from '../../redux-classes/actions/UserActions';
import {useDispatch} from 'react-redux';
import React, {useEffect, useState} from 'react';
import CommonButton from '../../common/components/CommonButton';
import {globalColors} from '../../constant/Colors';
import {WelcomeScreenStyle} from './WelcomeScreenStyle';

const {OSManager} = NativeModules;

interface WelcomeScreenProps {
  userFullName?: string;
  navigation?: any;
}

export const isValidUser = (userFullName: string): boolean => {
    if (userFullName.trim().length > 3) {
      return true;
    } else {
      return false;
    }
  };

/**
 * welcome screen 
 *
 * @param {*} {navigation}
 * @returns jsx
 */
const WelcomeScreen: React.FC<WelcomeScreenProps> = ({navigation}) => {

  const [userFullName, setUserFullName] = useState('');

  const dispatch = useDispatch();

  const checkTargetOS = async () => {
    const message = await OSManager.isAppRunningOnDevice();
    Alert.alert(message);
  };
  
  useEffect(() => {
    checkTargetOS();
  }, []);

  const onSavePress = () => {
    if (isValidUser(userFullName)) {
        dispatch(UserActions.saveUserInfo(userFullName));
        navigation?.navigate('ButtonScreen');
    } else {
      Alert.alert('User name should be atleast 3 character.');
    }
  };

  return (
    <SafeAreaView style={WelcomeScreenStyle.safeArea}>
      <View style={WelcomeScreenStyle.container}>
        <View style={WelcomeScreenStyle.userViewContainer}>
          <TextInput
            style={WelcomeScreenStyle.inputStyle}
            testID="User Name"
            placeholder="User Name"
            onChangeText={(userName) => setUserFullName(userName)}
            value={userFullName}
            autoCorrect={false}
            placeholderTextColor={'lightgrey'}
          />
        </View>

        <CommonButton
          onPress={() => onSavePress()}
          style={{backgroundColor: globalColors.sky}}
          testId="Save"
          titleString="Save"
          textStyle={WelcomeScreenStyle.saveButtonStyle}
        />
      </View>
    </SafeAreaView>
  );
};

export default WelcomeScreen;