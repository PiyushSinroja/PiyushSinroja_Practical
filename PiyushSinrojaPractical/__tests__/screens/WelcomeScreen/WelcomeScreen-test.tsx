import React from 'react';
import {Provider} from 'react-redux';
import WelcomeScreen from '../../../source/screens/WelcomeScreen/WelcomeScreen';
import {isValidUser} from '../../../source/screens/WelcomeScreen/WelcomeScreen';

import configureStore from '../../../source/redux-classes/ConfigStore';
const store = configureStore();
import {render, fireEvent} from '@testing-library/react-native';
import {Alert} from 'react-native';
describe('Welcome Screen', () => {
  const {toJSON, getByTestId, getByPlaceholderText} = render(
    <Provider store={store}>
      <WelcomeScreen />
    </Provider>,
  );

  
  describe('click on save', () => {
    it('check the user name', () => {
      const spy = jest.spyOn(Alert, 'alert');
      expect(isValidUser('Piyush')).toBeTruthy();
      expect(isValidUser('pi')).toBeFalsy();
      fireEvent.changeText(getByPlaceholderText('User Name'), '');
      fireEvent.press(getByTestId('Save'));
      expect(spy).toHaveBeenCalledTimes(1);

      fireEvent.changeText(getByTestId('User Name'), 'Piyush Sinroja');
      fireEvent.press(getByTestId('Save'));

      expect(getByTestId('User Name').props.value.length).toBeGreaterThanOrEqual(
        3,
      );
    });
  });
  expect(toJSON()).toMatchSnapshot();
});


describe('Validations', () => {
  it('check user name validation', () => {
    expect(isValidUser('hemant')).toBeTruthy();
    expect(isValidUser('he')).toBeFalsy();
  });
});
