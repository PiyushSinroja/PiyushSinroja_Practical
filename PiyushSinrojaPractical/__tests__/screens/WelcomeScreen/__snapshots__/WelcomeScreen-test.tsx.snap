// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "marginTop": 40,
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="User Name"
        placeholderTextColor="lightgrey"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "gray",
            "borderRadius": 5,
            "borderWidth": 1,
            "fontSize": 14,
            "height": 73.37,
            "justifyContent": "center",
            "marginEnd": 15,
            "marginStart": 15,
            "marginTop": 0,
            "paddingStart": 10,
          }
        }
        testID="User Name"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "marginBottom": 20,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#5AAAF2",
            "borderRadius": 5,
            "height": 80.03999999999999,
            "justifyContent": "center",
            "marginEnd": 20,
            "marginStart": 20,
            "opacity": 1,
          }
        }
        testID="Save"
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 17,
                "fontWeight": "normal",
              },
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
            ]
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
